core/api/wallet/ledger.cpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "api/wallet/ledger.hpp"
+
+#include <algorithm>
+#include "api/rpc/json.hpp"
+#include "codec/json/json.hpp"
+#include "common/bytes_cow.hpp"
+#include "common/error_text.hpp"
+#include "cpp-ledger/filecoin/ledger_filecoin.hpp"
+#include "primitives/address/address_codec.hpp"
+
+namespace fc::api {
+  using ledger::kPathLength;
+  using ledger::filecoin::LedgerFilecoinManager;
+  using primitives::address::decodeFromString;
+
+  const auto app_error = ERROR_TEXT("cannot open Ledger Filecoin App");
+
+  Ledger::Ledger(const MapPtr &store) : store(store) {}
+
+  bool Ledger::Has(const Address &address) const {
+    return store->contains(encode(address));
```

Comment:
Device master seed may change, so db cached address/path pair becomes invalid. `Has` may ask and compare address from device.  
Add `// note:` (or todo) about this.